server/src/test/java/io/crate/expression/scalar/object/ObjectKeysFunctionTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.object;
+
+import io.crate.expression.scalar.ScalarTestCase;
+import org.junit.Test;
+import java.util.List;
+
+
+public class ObjectKeysFunctionTest extends ScalarTestCase {
+
+    @Test
+    public void testNullInput() {
+        assertEvaluate("object_keys(null)", null);
+    }
+
+    @Test
+    public void testArrayInput() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Unknown function: object_keys(_array(1)), no overload found for"+
+                                        " matching argument types: (integer_array). Possible candidates:"+
+                                        " object_keys(object):array(text)");
+        assertEvaluate("object_keys([1])", null);
+    }
+
+    @Test
+    public void testEmptyObject() {
+        assertEvaluate("object_keys({})", List.of());
+    }
+
+    @Test
+    public void testObjectString() {
+        assertEvaluate("object_keys('{\"x\":1}')", List.of("x"));
+    }
+
+    @Test
+    public void testObjectLiteral() {
+        assertEvaluate("object_keys({a=1, b={c=2}})", List.of("a","b"));
```

Comment:
There is no need to add different tests for object string and object literal. this would not test the function but object string parsing.

Reply:
we can extract some value out of this test  - we can have at least one test with nested object structure just to demonstrate only top level keys selection

Reply:
I adjusted the naming and went with the nested / flat object tests